.github/workflows/build.yml

```diff
@@ -130,3 +74,359 @@ jobs:
+                'romac',
+                'settings_reset',
+                'tidbit'
+              ],
+              include: [
+                {
+                  "board": "bdn9_rev2"
+                },
+                {
+                  "board": "nice60"
+                },
+                {
+                  "board": "nrf52840_m2",
+                  "shield": "m60"
+                },
+                {
+                  "board": "planck_rev6"
+                },
+                {
+                  "board": "proton_c",
+                  "shield": "clueboard_california"
+                },
+                {
+                  "board": "nice_nano_v2",
+                  "shield": "kyria_left",
+                  "cmake-args": "-DCONFIG_ZMK_DISPLAY=y",
+                },
+                {
+                  "board": "nice_nano_v2",
+                  "shield": "kyria_right",
+                  "cmake-args": "-DCONFIG_ZMK_DISPLAY=y",
+                },
+                {
+                  "board": "nice_nano",
+                  "shield": "romac_plus",
+                  "cmake-args": "-DCONFIG_ZMK_RGB_UNDERGLOW=y -DCONFIG_WS2812_STRIP=y"
+                }
+              ]
+            };
+
+            let include = [];
+            
+            coreCoverage.boards.forEach(b => {
+              coreCoverage.shields.forEach(s => {
+                include.push({
+                  board: b,
+                  shield: s
+                });
+              });
+            });
+
+            return [...include, ...coreCoverage.include];
+  board-changes:
+    if: ${{ needs.get-changed-files.outputs.board-changes == 'true' }}
+    runs-on: ubuntu-latest
+    needs: [get-grouped-hardware, get-changed-files]
+    outputs:
+      boards-include: ${{ steps.boards-list.outputs.result }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Use Node.js
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14.x'
+      - name: Install js-yaml
+        run: npm install js-yaml
+      - uses: actions/github-script@v4
+        id: boards-list
+        with:
+          script: |
+            const fs = require('fs');
+            const yaml = require('js-yaml');
+
+            const changedFiles = JSON.parse(`${{ needs.get-changed-files.outputs.changed-files }}`);
+            const metadata = JSON.parse(`${{ needs.get-grouped-hardware.outputs.organized-metadata }}`);
+            const boardChanges = new Set(changedFiles.filter(f => f.startsWith('app/boards')).map(f => f.split('/').slice(0, 4).join('/')));
+
+            include = [];
+
+            for (const bc of boardChanges) {
+              const globber = await glob.create(bc + "/*.zmk.yml");
+              const files = await globber.glob();
+
+              const aggregated = files.flatMap((f) =>
+                yaml.loadAll(fs.readFileSync(f, "utf8"))
+              );
+
+              aggregated.forEach(hm => {
```

Comment:
Similar grumblings about using `forEach` here instead of mapping, but this also works fine.